src/Forte.Optimizely.ContentUsage/Api/Features/ContentUsage/ContentUsageController.cs

```diff
@@ -31,24 +31,49 @@ public ContentUsageController(IContentTypeRepository contentTypeRepository, Cont
     [SwaggerResponse(StatusCodes.Status200OK, null, typeof(IEnumerable<ContentUsageDto>))]
     [SwaggerResponse(StatusCodes.Status404NotFound)]
     [Route("[action]", Name = GetContentUsagesRouteName)]
-    public ActionResult GetContentUsages([FromQuery] [BindRequired] Guid guid)
+    public ActionResult GetContentUsages([FromQuery] GetContentUsagesQuery queryData)
     {
-        var contentType = _contentTypeRepository.Load(guid);
+        var contentType = _contentTypeRepository.Load(queryData.Guid);
 
         if (contentType == null) return NotFound();
 
-        var contentUsages = _contentUsageService.GetContentUsages(contentType);
+        var contentUsagesQuery = _contentUsageService.GetContentUsages(contentType);
 
-        var contentUsagesDto = contentUsages.Select(contentUsage => new ContentUsageDto
+        contentUsagesQuery = string.IsNullOrEmpty(queryData.Query)
+            ? contentUsagesQuery
+            : contentUsagesQuery.Where(x => x.Name.Contains(queryData.Query, StringComparison.InvariantCultureIgnoreCase));
+        
+        var contentUsages = contentUsagesQuery.ToArray();
+        
+        var currentPage = queryData.Page <= 0 ? 0 : queryData.Page - 1;
```

Comment:
Looks like this guard is redundant since `Skip()` applies the same check under the hood